python/test/unit/language/test_annotations.py

```diff
@@ -49,3 +49,14 @@ def _kernel(X: torch.Tensor, N: int, BLOCK_SIZE: tl.constexpr):
         _kernel[(1, )](x.shape[0], x.shape[0], 32)
     except AttributeError:
         pass
+
+
+# Test that float annotations are ignored
+def test_float_annotation(device):
+
+    @triton.jit
+    def _kernel(X: tl.float16):
+        tl.device_assert(X.dtype == tl.float32)
```

Comment:
Can we raise an error instead of ignoring it? That way it's not bc-breaking if we decide we want to support it down the line.

Reply:
Having `def kernel(a: tl.uint32, b: tl.float16)` where one annotation is functional and the other is silently dropped seems inconsistent to me. I think respecting all dtype annotations, but raising an error in cases we don't support yet gives a more consistent user experience.